src/ansi-c/builtin_factory.cpp

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************\
+
+Module:
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include "builtin_factory.h"
+#include "ansi_c_internal_additions.h"
+
+#include "ansi_c_parser.h"
+#include "ansi_c_typecheck.h"
+
+#include <cstring>
+#include <ostream>
+#include <sstream>
+
+const char windows_headers[]=
+  "int __noop();\n"
+  "int __assume(int);\n";
+
+//! Advance to the next line
+static const char *next_line(const char *line)
+{
+  const char *end=strchr(line, '\n');
+
+  if(end==NULL)
+    return strchr(line, 0);
+  else
+    return end+1;
+}
+
+//! Look for given pattern in given string.
+//! Add line with pattern to 'out' if found.
+//! \return 'true' if found
+static bool find_pattern(
+  const std::string &pattern,
+  const char *header_file,
+  std::ostream &out)
+{
+  // read line-by-line
+  const char *line, *line_end;
```

Comment:
How about using a `std::istringstream` here, and thus getting rid of all the C code? Roughly:
````c++
std::istringstream hdr(header_file);
std::string line;
while(std::getline(hdr, line))
{
  line = strip_string(line);
  if(has_prefix(line, "//") || line.find(pattern) == std::string::npos)
    continue;
  
  out << line;
  return true;
}

return false;
````

Reply:
Will compare speed; the C++ version does memory allocation.

Reply:
I'll stick to the C version. The C++ above is 27x slower.

Reply:
As per our Chime conversation I think that by itself is not sufficient data to make a judgement upon. I would favour readability and maintainability over optimisation here, but I don't intend to make that a blocker.